tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc

```diff
@@ -269,6 +301,34 @@ StatusOr<AutotuneResult> CudnnConvAlgorithmPicker::PickBestAlgorithmNoCache(
     return &stream_opt.value();
   }();
 
+  StatusOr<AutotuneResult> result_or(InternalError("Unknown platform."));
+  // Check StreamExecutor on which platform it is. ROCm and Cuda implementation
+  // have diverged. Secifically, we need to make sure redzone allocator related
+  // utilities are not used in ROCm routine
+  if (stream_exec_->platform_kind() == se::PlatformKind::kROCm) {
+    result_or = PickBestAlgorithmNoCacheRocm(*instr, allocator, stream);
+  } else if (stream_exec_->platform_kind() == se::PlatformKind::kCuda) {
+    result_or = PickBestAlgorithmNoCacheCuda(*instr, allocator, stream);
+  }
+
+  if (result_or.ok()) {
+    tensorflow::mutex_lock lock(autotune_cache_lock);
+    CHECK(autotune_cache.insert({key, result_or.ValueOrDie()}).second);
+  }
+  return result_or;
+}
+
+StatusOr<tensorflow::AutotuneResult>
+GpuConvAlgorithmPicker::PickBestAlgorithmNoCacheCuda(
+    const HloCustomCallInstruction& instr, se::DeviceMemoryAllocator* allocator,
```

Comment:
This is so that I'm aligned with cpp style guide in [Output Parameters](https://google.github.io/styleguide/cppguide.html#Output_Parameters) and [Reference Arguments](https://google.github.io/styleguide/cppguide.html#Reference_Arguments) sections.

`instr` is a strictly input argument therefore goes with const reference.

Reply:
Sure, but such refactorings need to be separate. If you would like to do that, please submit a separate PR, and then this PR will depend on it.

Reply:
Ok, I'd revert the refactoring changes in this PR, making it consistent with existing convention make it easier for me as well.